main.go

```diff
@@ -0,0 +1,333 @@
+// Web Chat in Go is a simple chat built with the websocket protocol (rfc6455)
+// on the back end and a free bootstrap template on the front end.
+package main
+
+import (
+	"bufio"
+	"fmt"
+	"html/template"
+	"net"
+	"net/http"
+	"time"
+
+	"web-chat-go/hub"
+	"web-chat-go/user"
+
+	"github.com/google/uuid"
+	"github.com/gorilla/websocket"
+	_ "github.com/lib/pq"
+	"golang.org/x/crypto/bcrypt"
+)
+
+type session struct {
+	un           string
+	lastActivity time.Time
+}
+
+var tpl *template.Template
+var upgrader = websocket.Upgrader{
+	// BufferSize determines how many bytes does the CPU handles each load,
+	// the bigger the size the less the CPU is going to work as less loads
+	// would need to be processed.
+	ReadBufferSize:  1024,
+	WriteBufferSize: 1024,
+}
+
+// UserName, User
+var dbUsers = make(map[string]user.User)
+
+// Session ID, UserName
+var dbSessions = make(map[string]session)
+
+// The single instance of "hub" is assigned.
+var h = hub.NewHub()
```

Comment:
I think you quoted me as saying a var name should be named appropriately for the scope of it's usage. h is too short for a global